packages/labs/compiler/README.md

```diff
@@ -1,6 +1,93 @@
 
-> **Warning**
-> Work in progress, not ready for external use.
+A compiler for optimizing Lit templates.
 
-Tracking RFC: https://github.com/lit/rfcs/pull/21
+> **Warning** `@lit-labs/compiler` is part of the Lit Labs set of packages – it is published
+> in order to get feedback on the design and may receive breaking changes.
+> Tracking RFC: https://github.com/lit/rfcs/pull/21
+
+> **Warning** `@lit-labs/compiler` is not yet published so it cannot be installed from npm.
+
+## Overview
+
+`@lit-labs/compiler` exports a [TypeScript
+Transformer](https://github.com/itsdouges/typescript-transformer-handbook#the-basics)
+that can be run over your JavaScript or TypeScript files to optimize away the
+`lit-html` **prepare** render phase. For template heavy applications this can result in a quicker first render.
+
+## Usage
+
+This transformer can be used anywhere TypeScript transformers are accepted, which is dependent on your build setup.
+
+Below is an example using [Rollup](https://rollupjs.org/) with the plugin [`@rollup/plugin-typescript`](https://www.npmjs.com/package/@rollup/plugin-typescript):
+
+```js
+// File: rollup.config.js
+import typescript from '@rollup/plugin-typescript';
+import {compileLitTemplates} from '@lit-labs/compiler';
+
+export default {
+  // ...
+  plugins: [
+    typescript({
+      transformers: {
+        before: [compileLitTemplates()],
+      },
+    }),
+    // other rollup plugins
+  ],
+};
+```
+
+See an example of the transformer in use in this project's test for source-maps validity in this [rollup config file](https://github.com/lit/lit/blob/main/packages/labs/compiler/rollup.source_map_tests.js).
+
+# FAQ
+
+## What are the tradeoffs for using the compiler?
+
+1. Running the compiler requires a build step that can accept a TypeScript transformer.
+
+2. The very first template render is faster (sometimes up to 45% faster for template heavy sites), but it currently comes with a 5% output file size increase (gzipped).
+
+## How do I know optimizations have been applied?
+
+Given your original source code containing the `html` tag function to declare templates:
+
+```js
+const hi = (name) => html`<h1>Hello ${name}!</h1>`;
+```
+
+This code should have been emitted at the end of your build without the `html` tag function.
+E.g. the above authored example is transformed into something like:
+
+```js
+const b = (s) => s;
+const lit_template_1 = {h: b`<h1>Hello <?></h1>`, parts: [{type: 2, index: 1}]};
+const hi = (name) => ({_$litType$: lit_template_1, values: [name]});
+```
+
+## What templates are optimized by the compiler?
+
+In order for a template to be optimized by the compiler, it must be:
+
+1. A well-formed template that wouldn't raise runtime diagnostics. For example, templates with expressions in [invalid locations](https://lit.dev/docs/templates/expressions/#invalid-locations) will not be compiled.
+1. Use `html` imported directly from the module `lit` or `lit-html`. The following imports are supported:
```

Comment:
Maybe clarify that re-exports are not supported

Reply:
Done!